Source/Core/Core/IOS/Network/SSL.cpp

```diff
@@ -35,6 +35,40 @@ static constexpr mbedtls_x509_crt_profile mbedtls_x509_crt_profile_wii = {
     0,         /* No RSA min key size */
 };
 
+namespace
+{
+// Dirty workaround to disable SNI which isn't supported by the Wii.
+//
+// This SSL extension can ONLY be disabled by undefining
+// MBEDTLS_SSL_SERVER_NAME_INDICATION and recompiling the library. When
+// enabled and if the hostname is set, it uses the SNI extension which is sent
+// with the Client Hello message.
+//
+// This workaround doesn't require to recompile the library. It does so by
+// defering mbedtls_ssl_set_hostname after the Client Hello message. The send
+// callback is used as it's the (only?) hook called at the beginning of
+// each step of the handshake by the mbedtls_ssl_flush_output function.
+//
+// The hostname needs to be set as it is checked against the Common Name field
+// during the certificate verification process.
+int mbedtls_net_send_without_sni(void* ctx, const unsigned char* buf, size_t len)
```

Comment:
The function name makes this look like it's an mbedtls function. Can we rename this to avoid any possible confusion? (Maybe `SendWithoutSNI`?)